go/libraries/doltcore/sqle/dtables/branches_table.go

```diff
@@ -76,6 +76,10 @@ func (bt *BranchesTable) PartitionRows(sqlCtx *sql.Context, part sql.Partition)
 	return NewBranchItr(sqlCtx, bt.ddb)
 }
 
+func (bt *BranchesTable) IsTemporary() bool {
```

Comment:
I'm thinking that IsTemporary should be on a separate TemporaryTable interface in gms

Reply:
I think there is less code with that approach. Although a separate interface feels wrong because every single table type can be a temporary table. I can see us in the future even allowing systems tables to be manipulated in temporary table formats to allow greater experimentation. 

Integrators should also be aware that they need to consider temporary table support.